Server/Quality_of_Life_changer/CalendarAdapter.cs

```diff
@@ -1,58 +1,130 @@
 ﻿using Google.Apis.Auth.OAuth2;
 using Google.Apis.Calendar.v3;
+using Google.Apis.Calendar.v3.Data;
 using Google.Apis.Services;
 using Quality_of_Life_changer.Contracts.Interfaces;
 using Quality_of_Life_changer.Model.Entities;
+using static Google.Apis.Auth.OAuth2.GoogleClientSecrets;
 
 namespace Quality_of_Life_changer.Adapter;
 
 public class CalendarAdapter : ICalendarAdapter
 {
     private static readonly string ApplicationName = "Quality of Life changer";
 
-    // If modifying these scopes, delete your previously saved credentials
-    // at ~/.credentials/calendar-dotnet-quickstart.json
-    private static readonly string[] Scopes = {CalendarService.Scope.CalendarReadonly};
-    private readonly CalendarService _calendarService;
+    private static readonly string[] CalendarScopes = {CalendarService.Scope.Calendar};
+
+    private CalendarService _calendarService;
+
+    private ClientSecrets _secrets;
 
     public CalendarAdapter()
     {
-        //UserCredential credential;
-        var credential =
-            GoogleCredential.FromStream(new FileStream("credentials.json", FileMode.Open, FileAccess.Read))
-                .CreateScoped(Scopes);
+        var fileName = GetFileName();
+
+        SetupSecrets(fileName);
+
+        SetupCalendar();
```

Comment:
you're not waiting for `SetupCalendar` to complete the task.
either call `.Wait()` or `.Result`